### PR TITLE
Issue#3396 Use doclets to ingest code snippets in javadoc

### PR DESCRIPTION
**Pull request by srnagar**
This change is to integrate an [open source doclet tool](https://github.com/jtulach/codesnippet4javadoc) to enable ingesting code snippets into javadocs. 

This includes changes to javadoc in applicationconfig and keyvault client libraries to use as examples. 

More details on how to use the tool is[ documented here](https://microsoft.sharepoint.com/teams/AzureDeveloperExperience/_layouts/OneNote.aspx?id=%2Fteams%2FAzureDeveloperExperience%2FShared%20Documents%2FLanguage%20-%20Java%2FAzure%20SDK%20-%20Language%20-%20Java&wd=target%28Client%20Libraries.one%7C1809F821-9CEB-9245-B57B-A3E75FB83065%2FJavaDoc%20Samples%7C48D63C28-F0BE-CF45-8CED-140E9469E260%2F%29)

### CODE REVIEW

**Comment by alzimmermsft** [comment]
Is it possible to have the doclet code snippets in the samples folder? I don't think code samples should live within the API package.

File: applicationconfig/client/pom.xml [hunk 1]
```diff
@@ -92,9 +92,28 @@
...
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-javadoc-plugin</artifactId>
+            <configuration>
+              <failOnWarnings>true</failOnWarnings>
+            </configuration>
+          </plugin>
+        </plugins>
+      </build>
+    </profile>
+  </profiles>
```
Comment (JonathanGiles) [inline applicationconfig/client/pom.xml]: Invert this - opt out in java-lts, rather than opt-in in jdk 8.

File: core/pom.xml [hunk 2]
```diff
@@ -76,12 +76,31 @@
...
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-javadoc-plugin</artifactId>
+            <configuration>
+              <failOnWarnings>true</failOnWarnings>
+            </configuration>
+          </plugin>
+        </plugins>
+      </build>
+    </profile>
+  </profiles>
```
Comment (JonathanGiles) [inline core/pom.xml]: There should be no need to do this here and further down the PR, it should be inherited appropriately.

File: eng/spotbugs-aggregate-report/pom.xml [hunk 3]
```diff
@@ -43,6 +43,7 @@
             <configuration>
               <sources>
                 <source>..\..\applicationconfig\client\src\main\java</source>
+                <source>..\..\applicationconfig\client\src\samples\java</source>
                 <source>..\..\core\azure-core\src\main\java</source>
                 <source>..\..\core\azure-core-auth\src\main\jasva</source>
```
Comment (JonathanGiles) [inline eng/spotbugs-aggregate-report/pom.xml]: Can you fix the 'jasva' typo I spotted on this line?

Comment (srnagar) [reply eng/spotbugs-aggregate-report/pom.xml]: Done

File: core/pom.xml (hunk 2 above)
Comment (srnagar) [reply core/pom.xml]: Done

File: applicationconfig/client/pom.xml (hunk 1 above)
Comment (srnagar) [reply applicationconfig/client/pom.xml]: Done

**Pull request merged**